### DISCUSSION TITLE
Adding AUTO_INCREMENT to a column that is not a primary-key

### DISCUSSION THREAD

**Discussion by davnat** [sqlalchemy/sqlalchemy#10921] [Usage Questions] [answered]
In MariaDB documentation I read that AUTO_INCREMENT can be set in a column that is not necessarily the PRIMARY KEY:
 https://mariadb.com/kb/en/auto_increment/

Unfortunately, I couldn't find a way to express this using SQLAlchemy 1.4.47, i. e.: this does NOT work:
```python
class Tool(Base):
    tool_id = Column(
        INTEGER(11),
        nullable=False,
        unique=True,
        autoincrement=True,
    )
    tool_sn = Column(String(20), primary_key=True, nullable=False)
    tool_version = Column(String(50), primary_key=True, nullable=False)
    creation_date = Column(DateTime, nullable=False)
```
Can anyone help me?

Thank you, cheers.

**Comment by zzzeek** [comment, marked as answer]
SQLAlchemy only applies AUTO_INCREMENT to primary key columns.     There's not a hook to customize this so simply use [ALTER TABLE](https://mariadb.com/kb/en/alter-table/) instead; using [DDL](https://docs.sqlalchemy.org/en/20/core/ddl.html):

```py
from sqlalchemy import DDL, event


event.listen(
    Tool.__table__, 'after_create', 
    DDL("ALTER TABLE tool MODIFY tool_id INT(11) AUTO_INCREMENT").execute_if(dialect='mariadb')
)

```

**Reply by davnat** [reply]
This works, thanks.
